### ISSUE TITLE
Expose parseRedirectURL from signInWithRedirect to allow it manually be called

### ISSUE THREAD

**Issue by JanStevens** [aws-amplify/amplify-js#12737]
### Is this related to a new or existing framework?

React, Next.js

### Is this related to a new or existing API?

Authentication

### Describe the feature you'd like to request

Hi,

Because of the code splitting / tree shaking it's possible that the [`parseRedirectUrl`](https://github.com/aws-amplify/amplify-js/blob/1477488a1124459d0594ab96e90140201a3b15d4/packages/auth/src/providers/cognito/apis/signInWithRedirect.ts#L491) will never be called after successfully logging in with `signInWithRedirect`.

In Next.js this can happen easily since it automatically code split per page, for example I have a login page at `/account/login` which has an `signInWithRedirect({ provider: 'apple' })` button, my configured `redirectSignIn` points to `/account/redirecting` where I show a loading indicator. On the redirect page I don't include the `signInWithRedirect` so the Hub.listen will never be called causing the `parseRedirectUrl` to be never called.

### Describe the solution you'd like

Expose `parseRedirectURL` so it can be manually called in a useEffect, for example:

```typescript
useEffect(() => {
  parseRedirectURL()
}, [])
```

### Describe alternatives you've considered

Right now the only solution I have is to import `signInWithRedirect` and use it somewhere so it does not get removed, a bit ugly

```typescript
// This file needs to be included so all the listeners are setup correctly
import { signInWithRedirect } from 'aws-amplify/auth';

const willNeverBeCalledButNeeded = () => signInWithRedirect
``` 

### Is this something that you'd be interested in working on?

- [X] 👋 I may be able to implement this feature request

**Comment by nadetastic** [comment]
Related - https://github.com/aws-amplify/amplify-js/issues/12589

**Comment by cwomack** [comment]
@JanStevens, thank you for opening this issue.  We'll mark this as a feature request for now and will review this internally!

**Comment by nadetastic** [comment]
Hi @JanStevens Following up here - with the latest version of aws-amplify, this feature request has be addressed. For additional information on how, please take a look at the documentation here - [docs.amplify.aws/react/build-a-backend/auth/add-social-provider/#required-for-multi-page-applications-complete-social-sign-in-after-redirect](https://docs.amplify.aws/react/build-a-backend/auth/add-social-provider/#required-for-multi-page-applications-complete-social-sign-in-after-redirect)

In short, you will need to import the oauth listener to the page that you are redirecting to if it is not the same page that `signInWithRedirect()` is invoke from.

```js
import 'aws-amplify/auth/enable-oauth-listener';
```
Let me know if you have any questions.
